narwhal/primary/src/tests/primary_tests.rs

```diff
@@ -345,7 +345,7 @@ async fn test_request_vote_has_missing_parents() {
     let round_2_missing = round_2_certs[(NUM_PARENTS / 2)..].to_vec();
 
     // Create a test header.
-    let test_header = Header::V1(
+    let test_header = Header::V2(
```

Comment:
I also already have HeaderV2 migrations for this file and synchronizer_tests too in my existing PR :/ I appreciate the attention to keeping interfaces clean, at the same time it might have been nicer if you had waited to do this or checked first. now at least one of us is going to have a time consuming merge on our hands.


rather than construct with `Header::V2(..builder_output..)` I think it's cleaner to just call `into`? then the code doesn't need to change yet again if we make a `HeaderV3`. that's what I did in mine anyway

Reply:
Cool. I missed these changes during review. I can definitely revert changes in these two files. I can revert most of the HeaderV1 cleanups in other files too if they are being taken care of, and just keep the `types/src/primary.rs` cleanups.